src/diffusers/models/molecule_gnn.py

```diff
@@ -0,0 +1,640 @@
+        else:
+            self.activation = None
+        if dropout:
+            self.dropout = nn.Dropout(dropout)
+        else:
+            self.dropout = None
+
+        self.layers = nn.ModuleList()
+        for i in range(len(self.dims) - 1):
+            self.layers.append(nn.Linear(self.dims[i], self.dims[i + 1]))
+
+    def forward(self, input):
+        """"""
+        x = input
+        for i, layer in enumerate(self.layers):
+            x = layer(x)
+            if i < len(self.layers) - 1:
+                if self.activation:
+                    x = self.activation(x)
+                if self.dropout:
+                    x = self.dropout(x)
+        return x
+
+
+class ShiftedSoftplus(torch.nn.Module):
+    def __init__(self):
+        super(ShiftedSoftplus, self).__init__()
+        self.shift = torch.log(torch.tensor(2.0)).item()
+
+    def forward(self, x):
+        return F.softplus(x) - self.shift
+
+
+class CFConv(MessagePassing):
+    def __init__(self, in_channels, out_channels, num_filters, nn, cutoff, smooth):
+        super(CFConv, self).__init__(aggr="add")
+        self.lin1 = Linear(in_channels, num_filters, bias=False)
+        self.lin2 = Linear(num_filters, out_channels)
+        self.nn = nn
+        self.cutoff = cutoff
+        self.smooth = smooth
+
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        torch.nn.init.xavier_uniform_(self.lin1.weight)
+        torch.nn.init.xavier_uniform_(self.lin2.weight)
+        self.lin2.bias.data.fill_(0)
+
+    def forward(self, x, edge_index, edge_length, edge_attr):
+        if self.smooth:
+            C = 0.5 * (torch.cos(edge_length * np.pi / self.cutoff) + 1.0)
+            C = C * (edge_length <= self.cutoff) * (edge_length >= 0.0)  # Modification: cutoff
+        else:
+            C = (edge_length <= self.cutoff).float()
+        W = self.nn(edge_attr) * C.view(-1, 1)
+
+        x = self.lin1(x)
+        x = self.propagate(edge_index, x=x, W=W)
+        x = self.lin2(x)
+        return x
+
+    def message(self, x_j, W):
+        return x_j * W
+
+
+class InteractionBlock(torch.nn.Module):
+    def __init__(self, hidden_channels, num_gaussians, num_filters, cutoff, smooth):
+        super(InteractionBlock, self).__init__()
+        mlp = Sequential(
+            Linear(num_gaussians, num_filters),
+            ShiftedSoftplus(),
+            Linear(num_filters, num_filters),
+        )
+        self.conv = CFConv(hidden_channels, hidden_channels, num_filters, mlp, cutoff, smooth)
+        self.act = ShiftedSoftplus()
+        self.lin = Linear(hidden_channels, hidden_channels)
+
+    def forward(self, x, edge_index, edge_length, edge_attr):
+        x = self.conv(x, edge_index, edge_length, edge_attr)
+        x = self.act(x)
+        x = self.lin(x)
+        return x
+
+
+class SchNetEncoder(Module):
+    def __init__(
+        self, hidden_channels=128, num_filters=128, num_interactions=6, edge_channels=100, cutoff=10.0, smooth=False
```

Comment:
Isn't paper default for `num_interactions` 3?

Reply:
For the SchNet paper? No clue. This was copied from author.

Reply:
Ok, I guess we can close this comment then (it is 3 in the SchNet paper, I looked it up along your code yesterday, but it's not that important, and if the other model relies on 6 as default, might be easier to keep 6)